ffi/dart/hook/build.dart

```diff
@@ -173,20 +174,35 @@ final class LocalMode extends BuildMode {
 }
 
 final class CheckoutMode extends BuildMode {
-  final Uri? checkoutPath;
+  late final Uri checkoutPath;
 
-  CheckoutMode(super.input, this.checkoutPath);
-
-  @override
-  Future<Uri> build() async {
+  CheckoutMode(super.input, Uri? checkoutPath) {
     print('Running in `checkout` mode');
     if (checkoutPath == null) {
       throw ArgumentError(
         'Specify the ICU4X checkout folder with the `checkoutPath` key in your '
         'pubspec build options.',
       );
     }
-    if (!File.fromUri(checkoutPath!.resolve('Cargo.lock')).existsSync()) {
+    if (checkoutPath.path.contains('\$PUB_CACHE')) {
+      final envVars = Platform.environment;
```

Comment:
The environment variables are filtered for build hooks IIRC, so that will probably not work.

Reply:
That's surprising to me - https://github.com/dart-lang/native/blob/e0cd7d4555491d96a784ad362c569be1b53fa2f7/pkgs/hooks_runner/lib/src/build_runner/build_runner.dart#L545 does not list the pub cache env var. 

@dcharkes any idea why this is not filitered?

Reply:
This should not be accessible indeed. I have no idea why it is not filtered.